src/kudu/consensus/leader_election.cc

```diff
@@ -307,34 +328,417 @@ Status FlexibleVoteCounter::RegisterVote(
   return s;
 }
 
-std::pair<bool, bool> FlexibleVoteCounter::GetQuorumState() const {
+int FlexibleVoteCounter::FetchVotesRemainingInRegion(
+    const std::string& region) const {
+  const std::map<std::string, int>::const_iterator regional_yes_count =
+      yes_vote_count_.find(region);
+  const std::map<std::string, int>::const_iterator regional_no_count =
+      no_vote_count_.find(region);
+  const std::map<std::string, int>::const_iterator total_region_count =
+      voter_distribution_.find(region);
+
+  // All the following must at least be initialized to zero in the
+  // constructor.
+  CHECK(total_region_count != voter_distribution_.end());
+  CHECK(regional_yes_count != yes_vote_count_.end());
+  CHECK(regional_no_count != no_vote_count_.end());
```

Comment:
As long as you determine we are protected from non-validated input for `region`, better to write this logic like this, here and elsewhere:

```suggestion
  int regional_yes_count = FindOrDie(yes_vote_count_, region);
  int regional_no_count = FindOrDie(no_vote_count_, region);
  int total_region_count = FindOrDie(voter_distribution_, region);
```